src/red-hat-openshift.py

```diff
@@ -0,0 +1,105 @@
+import re
+import subprocess
+from pathlib import Path
+from common import endoflife
+from hashlib import sha1
+from os.path import exists
+from subprocess import call
+from typing import Union
+
+"""
+Fetch Red Hat OpenShift versions from the documentation git repository
+"""
+
+PRODUCT = "red-hat-openshift"
+REPO_URL = "https://github.com/openshift/openshift-docs.git"
+
+
+class Git:
+    """
+    Git cli wrapper
+    """
+
+    def __init__(self, url: str):
+        self.url: str = url
+        self._url_sha1: str = sha1(self.url.encode()).hexdigest()
+        self.repo_dir: Path = Path(f"~/.cache/git/red-hat-openshift_{self._url_sha1}").expanduser()
+
+    def run(self, cmd: str, return_output: bool = False) -> Union[bool, list]:
+        """
+        Run git command and return True on success or False on failure.
+        Optionaly returns command result instead.
+        """
+        git_opts = f"--git-dir={self.repo_dir}/.git --work-tree={self.repo_dir}"
+
+        if return_output:
+            child = subprocess.Popen(
+                f"git {git_opts} {cmd}",
+                shell=True,
+                stdout=subprocess.PIPE,
+            )
+            return child.communicate()[0].decode("utf-8").split("\n")
+
+        return call(f"git {git_opts} {cmd}",
+                    shell=True,
+                    stdout=subprocess.DEVNULL,
+                    stderr=subprocess.DEVNULL) == 0
+
+    def clone(self):
+        self.repo_dir.mkdir(parents=True, exist_ok=True)
+        if not exists(f"{self.repo_dir}/.git"):
+            self.run("init")
+            self.run(f"remote add origin {self.url}")
+
+    def list_branches(self):
+        raw = self.run(
+            "ls-remote origin 'refs/heads/enterprise-[4-9]*'",
```

Comment:
Why limiting to OpenShift 4+ ?

Reply:
They changed the filename schema from `ocp_{version}_release_notes.adoc` to `ocp-{version}-release-notes.adoc`. OpenShift 1-3 won't get any updates anymore, so it doesn't seemed to make sense to support that..

Reply:
Also, the content seems to be different in the old versions.

Reply:
Ok, thanks for the precision. Could be interesting to leave a comment about that in the script.

Reply:
good idea, added 05d22b601222bcca422fff07bcec192c06d30e83

Reply:
Thanks !